main.tf

```
@@ -8,14 +8,13 @@ locals {
   waf_v2_policies                        = local.enabled && length(var.waf_v2_policies) > 0 ? { for policy in flatten(var.waf_v2_policies) : policy.name => policy } : {}
   dns_firewall_policies                  = local.enabled && length(var.dns_firewall_policies) > 0 ? { for policy in flatten(var.dns_firewall_policies) : policy.name => policy } : {}
   network_firewall_policies              = local.enabled && length(var.network_firewall_policies) > 0 ? { for policy in flatten(var.network_firewall_policies) : policy.name => policy } : {}
-  assumed_arn                            = var.is_destroy ? var.firewall_manager_administrator_arn : var.organization_management_arn
 }
 
 data "aws_caller_identity" "default" {
   count = local.enabled ? 1 : 0
 }
 
 resource "aws_fms_admin_account" "default" {
-  provider   = aws.variable_arn
+  provider = aws.dynamic_arn
```

```suggestion
  provider = aws.admin_account
```

`dynamic_arn` does not convey why we need another provider.

---

admin_account does not describe that it will fail to delete though.
This resource requires admin_account to create, but the Account its created in to delete. thus requiring a provider whos arn changes based on the operation you are performing

---

we can all it `dynamic_provider` (not `dynamic_arn`).
`arn` is not a provider that goes into `provider` filed.

Also, I did not look into that yet, but how are you planning to specify different providers (or diff ARNs) when you create and when you destroy?

---

@Benbentwo  please resolve the conflicts

---

```
locals {
  assumed_arn = var.is_destroy ? var.firewall_manager_administrator_arn : var.organization_management_arn
}

module "components_firewall_manager" {
  providers = {
    aws.dynamic_arn = aws.dynamic_arn
    aws = aws
  }
...
}
```

Unfortunately we don't have a hook to set `is_destroy` based on the terraform command being ran